src/filters.ts

```diff
@@ -191,27 +256,63 @@ export class MatchNestedBundle implements ItemFilter {
  *
  *   { never: true }
  */
-export function createFilter(filter: any): ItemFilter {
+export function createFilter(
+  filter: any,
+  logger: Logger = defaultLogger,
+): ItemFilter {
+  const log = logger.child({ function: 'createFilter' });
+
   if (filter === undefined || filter === '') {
-    return new NeverMatch();
+    return new NeverMatch(log);
   }
 
   if (filter?.tags) {
-    return new MatchTags(filter.tags);
+    return new MatchTags(
+      filter.tags,
+      log.child({ itemFilter: JSON.stringify({ tags: filter.tags }) }),
+    );
   } else if (filter?.attributes) {
-    return new MatchAttributes(filter.attributes);
+    return new MatchAttributes(
+      filter.attributes,
+      log.child({
+        itemFilter: JSON.stringify({ attributes: filter.attributes }),
+      }),
+    );
   } else if (filter?.isNestedBundle) {
-    return new MatchNestedBundle();
+    return new MatchNestedBundle(
+      log.child({
+        itemFilter: JSON.stringify({ isNestedBundle: filter.isNestedBundle }),
+      }),
+    );
   } else if (filter?.not) {
-    return new NegateMatch(createFilter(filter.not));
+    const childLogger = log.child({
+      itemFilter: JSON.stringify({ not: filter.not }),
+    });
+    return new NegateMatch(createFilter(filter.not, childLogger), childLogger);
   } else if (filter?.and) {
-    return new MatchAll(filter.and.map(createFilter));
+    const childLogger = log.child({
+      itemFilter: JSON.stringify({ and: filter.and }),
+    });
+    return new MatchAll(
+      filter.and.map((and: any) => createFilter(and, childLogger)),
```

Comment:
I think we could make this a bit nicer if we create a filter "path" that was passed along and included in the log context. The top level `itemFilterPath` would be `[]`. When a filter with other nested filters is create the filter at that level would be appended (be sure to create a copy of the array when doing this!). Then when each child logger is created, the current path + the filter in question is added. In the end you'd get paths like this `[and, or, or, tags]`.

Reply:
I see what you mean, I'll implement it and we'll see how it turns out.